networkx/convert_matrix.py

```diff
@@ -635,18 +636,19 @@ def _csc_gen_triples(A):
 
     """
     ncols = A.shape[1]
-    data, indices, indptr = A.data, A.indices, A.indptr
-    for i in range(ncols):
-        for j in range(indptr[i], indptr[i + 1]):
-            yield int(indices[j]), i, data[j]
+    indptr, src_indices, data = A.indptr, A.indices, A.data
```

Comment:
Reverted in the last commit to see (and show) what happens.

Reply:
Actually, the `*_gen_triples` functions are only used by `_generate_weighted_edges`, which is only used by `from_biadjacency_matrix` and `from_scipy_sparse_array`, which create _graphs_, not _arrays_.

Suppose we wanted these graphs to use numpy scalars so they can be converted back into arrays with the same dtype. Would this be worth it? I would still be against it, because it's inconsistent with other graph creation functions, and `to_scipy_sparse_array` and `biadjacency_matrix` accept dtype arguments to make this explicit.

I'm going to unrevert my reversion. Please let me know if I misunderstood something and ought to make a change.